buildSrc/src/main/groovy/BuildkitBuild.groovy

```diff
@@ -0,0 +1,358 @@
+                imageRepo.add([name: targetRepo, tag: targetTags])
+            } else {
+                imageRepo.add([name: targetRepo, tag: ""])
+            }
+        } else if (dockerHubPublish.get()) {
+            targetRepo = "corda"
+            targetTags = ["${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (preTest.get()) {
+            targetRepo = "corda-os-docker-pre-test.software.r3.com"
+            targetTags = ["preTest-${tagPrefix}${version}", "preTest-${tagPrefix}${gitRevision}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'RC' || releaseType == 'GA') {
+            targetRepo = "corda-os-docker-stable.software.r3.com"
+            targetTags = ["${tagPrefix}latest", "${tagPrefix}${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'BETA' && !nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-unstable.software.r3.com"
+            targetTags = ["${tagPrefix}unstable", "${tagPrefix}${gitRevision}", "${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'ALPHA' && !nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-dev.software.r3.com"
+            targetTags = ["${tagPrefix}${gitRevision}", "${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'BETA' && nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-nightly.software.r3.com"
+            targetTags = ["${tagPrefix}nightly", "${tagPrefix}nightly-${timeStamp}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'ALPHA' && nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-nightly.software.r3.com"
+            targetTags = ["${tagPrefix}nightly-${version}", "${tagPrefix}nightly-${gitRevision}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else {
+            targetRepo = "corda-os-docker-dev.software.r3.com"
+            targetTags = ["latest-local", "${version}", "${gitRevision}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        }
+
+        List<String> javaArgs = new ArrayList<String>(arguments.get())
+        javaArgs.add("-Dlog4j2.debug=\${ENABLE_LOG4J2_DEBUG:-false}")
+
+        def baseImageName = "${baseImageTag.get().empty ? baseImageName.get() : "${baseImageName.get()}:${baseImageTag.get()}"}"
+        def entryName = overrideEntryName.get().empty ? projectName : overrideEntryName.get()
+
+        dockerAuth.add(targetRepo)
+
+        DockerLogin(dockerAuth)
+
+        String[] baseCommand
+        String[] opts
+        String[] commandTail
+
+        workingDir project.rootDir
+
+        for (repo in imageRepo) {
+            for (tag in repo.tag) {
+                if (isBuildx.get()) {
+                    CheckForBuildxContainer()
```

Comment:
you likley only need to hit this logic once not in your nested for loops where it will be triggered multiple times. I'd just pull this out of the loops entirely and do 

        if (isBuildx.get()) {
            CheckForBuildxContainer()
         }